articles/node_lifecycle.md

```diff
@@ -0,0 +1,287 @@
+run. This provides the node with a chance to clean up resources it was using.
+
+When a node is in this state, roslaunch may choose to reactivate the node and
+hope the error has gone away (accomplished by clearing the error and then going
+through the standard activation process), restart the node (and hope the error
+does not repeat), or it may choose to replace it with an alternative
+implementation that provides the same functionality. Restarting or replacing
+the node can be accomplished destroying it and creating a new instance of its
+replacement.
+
+### Destroyed state
+
+This is the state that a node goes through on its way to no longer existing.
+
+A node must be deactivated before it can be destroyed.
+
+On entering this state, the node's `onDestroyed` method must be run. This is
+the node's last chance to clean up.
+
+Implementation note: In object-oriented languages, the onDestroyed method can
+be mapped to the class destructor.
+
+
+
+
+## Interface
+
+A node must provide the messages and services defined in this section to
+qualify as a managed node. These services define the external interface of a
+managed node as seen by tools that perform the managing. As such, it should be
+possible to consider any object that provides these services correctly to be a
+managed node.
+
+Conversely, any object that provides these services but does not behave in the
+way defined in the life cycle state machine is malformed.
+
+These services may be provided via attributes and method calls (for local
+management) or via ROS messages and services (for remote management).
+In the case of providing a ROS middleware interface, specific topics must be
+used, and they should be placed in a suitable namespace.
+
+### Initialisation
+
+The following service should be called to initialise the node. This service must
+be named `initialise`.
+
+{% raw %}
+    ---
+    bool success
+    string result_msg
+{% endraw %}
+
+
+### Activation
+
+The following service should be called to activate the node. This service must
+be named `activate`.
+
+{% raw %}
+    ---
+    bool success
+    string result_msg
+{% endraw %}
+
+
+### Deactivation
+
+The following service should be called to deactivate the node. This service
+must be named `deactivate`.
+
+{% raw %}
+    ---
+    bool success
+    string result_msg
+{% endraw %}
+
+
+### Clearing the error state
+
+The following service should be called to clear the node's fatal error status.
+This is done when the node is to be re-activated in the hope that the error has
+gone away.
+
+{% raw %}
+    ---
+    bool success
+{% endraw %}
+
+### State reporting
+
+The following service should be called to query the current state of the node.
+The service must be named `get_state`.
+
+{% raw %}
+    ---
+    int8 created=0
+    int8 inactive=1
+    int8 active=2
+    int8 error=3
+    int8 destroyed=4
+
+    int8 state
+{% endraw %}
+
+The value of `state` must be one of the constants defined in the service
```

Comment:
Yes this is a change from the current implementation however it fits the anonymous publish/subscribe datamodel much better. In the case that there's only one publisher and one subscriber it doesn't make a difference. However in the case that you have two or more subscribers it does not necessarily make sense for all of them to be forced to be latched. For example if you want to just bring up a diagnostic display you may want the last value or you may want not want to waste the resources requesting the old data and be willing to wait for the new data. 

The publisher should not need to know the models of consumption of the downstream subscribers, it should just make the capability available that they can then request it. If publishers are latched and you don't want to incur a latched copy over wireless your best solution is to create a republisher topic locally which subscribes to the latched topic and republishes it unlatched. 

The current ROS implementation of latched topics was considered for the subscriber but was deemed too complicated and the simpler implementation was done on the publisher side to be able to support latching at some level. Previously we hadn't had that capability and it turned out to be very valuable. Looking back it might have been worth the effort to have supported it on the subscriber side, but at the time of implementation we had not learned how valuable it would be. Having learned how valuable latching is we'd like to build in the full implementation instead of following the previous implementation required by the pre-existing design decisions. 

@gbiggs It might be interesting to have two automatically available services or options to the service with semantics "get last" and "get next" to support the polling type interface too.